ha/k8s_setup/ha_setup.py

```diff
@@ -0,0 +1,337 @@
+            with open(const.HA_CONFIG_FILE, 'w+') as conf_file:
+                yaml.dump(conf_file_dict, conf_file, default_flow_style=False)
+
+            # First populate the ha.conf and then do init. Because, in the init, this file will
+            # be stored in the confstore as key values
+            ConfigManager.init("ha_setup")
+            self._confstore = ConfigManager.get_confstore()
+            Log.info(f'Populating the ha config file with consul_endpoint: {consul_endpoint}, \
+                      prometheus_endpoint:')
+
+            Log.info(f'Performing event_manager subscription')
+            event_manager = EventManager.get_instance()
+            event_manager.subscribe(const.EVENT_COMPONENT, [SubscribeEvent(const.POD_EVENT, ["online", "failed"])])
+            Log.info(f'event_manager subscription for {const.EVENT_COMPONENT}\
+                       is successful for the event {const.POD_EVENT}')
+
+            Log.info("config command is successful")
+            sys.stdout.write("config command is successful.\n")
+        except TypeError as type_err:
+            sys.stderr.write(f'HA config command failed: Type mismatch: {type_err}.\n')
+        except yaml.YAMLError as exc:
+            sys.stderr.write(f'Ha config failed. Invalid yaml configuration: {exc}.\n')
+        except OSError as os_err:
+            sys.stderr.write(f'HA Config failed. OS_error: {os_err}.\n')
+        except Exception as c_err:
+            sys.stderr.write(f'HA config command failed: {c_err}.\n')
+
+class InitCmd(Cmd):
+    """
+    Init Setup Cmd
+    """
+    name = "init"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process init command.
+        """
+        sys.stdout.write('HA initialization is done.\n')
+
+class TestCmd(Cmd):
+    """
+    Test Cmd
+    """
+    name = "test"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process test command.
+        """
+        sys.stdout.write('Yet to be implemented...\n')
+
+class ResetCmd(Cmd):
+    """
+    Reset Cmd
+    """
+    name = "reset"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process reset command.
+        """
+        sys.stdout.write('HA reset is yet to be implemented...\n')
+
+class CleanupCmd(Cmd):
+    """
+    Cleanup Cmd
+    """
+    name = "cleanup"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process cleanup command.
+        """
+        Log.info("Processing cleanup command")
+        try:
+            # Delete the config file
+            self.remove_config_files()
```

Comment:
Please remove the consul entries as well

Reply:
Done